mica-webapp/src/main/java/org/obiba/mica/web/controller/SearchController.java

```diff
@@ -1,15 +1,56 @@
 package org.obiba.mica.web.controller;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Maps;
+import com.google.common.eventbus.Subscribe;
+import org.obiba.mica.micaConfig.event.MicaConfigUpdatedEvent;
+import org.obiba.mica.micaConfig.service.MicaConfigService;
+import org.springframework.scheduling.annotation.Async;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.servlet.ModelAndView;
 
+import javax.inject.Inject;
+import java.util.Map;
+
 @Controller
 public class SearchController {
 
+  private MicaConfigService micaConfigService;
+
+  private String cachedMicaConfigJson = null;
+
+  @Inject
+  public SearchController(MicaConfigService micaConfigService) {
```

Comment:
Why not using the MicaConfigInterceptor where the MicaConfig object is already extracted and cached?

Reply:
I updated my branch and removed the cache.